internal/juju/jaas.go

```diff
@@ -0,0 +1,87 @@
+// Copyright 2024 Canonical Ltd.
+// Licensed under the Apache License, Version 2.0, see LICENCE file for details.
+
+package juju
+
+import (
+	"github.com/canonical/jimm-go-sdk/v3/api"
+	"github.com/canonical/jimm-go-sdk/v3/api/params"
+	jujuapi "github.com/juju/juju/api"
+
+	"github.com/juju/errors"
+)
+
+type jaasClient struct {
+	SharedClient
+	getJaasApiClient func(jujuapi.Connection) JaasAPIClient
+}
+
+func newJaasClient(sc SharedClient) *jaasClient {
+	return &jaasClient{
+		SharedClient: sc,
+		getJaasApiClient: func(conn jujuapi.Connection) JaasAPIClient {
+			return api.NewClient(conn)
+		},
+	}
+}
+
+// AddRelations attempts to create the provided slice of relationship tuples.
+func (jc *jaasClient) AddRelations(tuples []params.RelationshipTuple) error {
+	conn, err := jc.GetConnection(nil)
+	if err != nil {
+		return err
+	}
+	defer func() { _ = conn.Close() }()
+	cl := jc.getJaasApiClient(conn)
+	req := params.AddRelationRequest{
+		Tuples: tuples,
+	}
+	return cl.AddRelation(&req)
+}
+
+// DeleteRelations attempts to delete the provided slice of relationship tuples.
+func (jc *jaasClient) DeleteRelations(tuples []params.RelationshipTuple) error {
```

Comment:
issue: The jaasClient method should have unique input and output structures to avoid leaking jaas into the provider code and keep the separation of concerns. We've been cleaning up juju in this area.